spec/tbd_tests_spec.rb

```diff
@@ -2251,4 +2251,218 @@
+    expect(model.empty?).to be(false)
+    model = model.get
+
+    # Unaltered template 'FullServiceRestaurant' OpenStudio model (no
+    # constructions, no setpoints), to be modified by BTAP.
+    loops     = TBD.airLoopsHVAC?(model)
+    setpoints = TBD.heatingTemperatureSetpoints?(model)
+    setpoints = TBD.coolingTemperatureSetpoints?(model) || setpoints
+    expect(model.getConstructions.empty?).to be(true)
+    expect(setpoints).to be(false)
+    expect(loops).to be(false)
+
+    argh[:option] = "code (Quebec)"
+    json = TBD.process(model, argh)
+    expect(json.is_a?(Hash)).to be(true)
+    expect(json.key?(:io)).to be(true)
+    expect(json.key?(:surfaces)).to be(true)
+    io       = json[:io      ]
+    surfaces = json[:surfaces]
+    expect(TBD.error?).to be(true)
+    expect(TBD.logs.empty?).to be(false)
+    expect(io.nil?).to be(false)
+    expect(io.is_a?(Hash)).to be(true)
+    expect(io.empty?).to be(false)
+    expect(io.key?(:edges)).to be(false)
+    expect(surfaces.nil?).to be(false)
+    expect(surfaces.is_a?(Hash)).to be(true)
+    expect(surfaces.size).to eq(18)
+
+    # No constructions to derate - 'surfaces' only holds pre-TBD attributes.
+    surfaces.values.each do |surface|
+      expect(surface.is_a?(Hash)).to be(true)
+      expect(surface.key?(:space)).to be(true)
+      expect(surface.key?(:stype)).to be(true) # spacetype
+      expect(surface.key?(:conditioned)).to be(true)
+      expect(surface.key?(:deratable)).to be(true)
+      expect(surface.key?(:construction)).to be(false)
+      expect(surface[:conditioned]).to be(true)
+      expect(surface[:deratable]).to be(false)
+    end
+
+    # Fetch conditioned attic floor.
+    id    = "attic-floor-dinning"
+    expect(surfaces.key?(id)).to be(true)
+    space = surfaces[id][:space]
+    expect(space.partofTotalFloorArea).to be(false)
+    expect(space.thermalZone.empty?).to be(false)
+    zone  = space.thermalZone.get
+    expect(zone.isPlenum).to be(false)
+
+    heat = TBD.maxHeatScheduledSetpoint(zone)
+    cool = TBD.minCoolScheduledSetpoint(zone)
+    expect(heat.nil?).to be(false)
+    expect(cool.nil?).to be(false)
+    expect(heat.is_a?(Hash)).to be(true)
+    expect(cool.is_a?(Hash)).to be(true)
+    expect(heat.key?(:spt)).to be(true)
+    expect(cool.key?(:spt)).to be(true)
+    expect(heat.key?(:dual)).to be(true)
+    expect(cool.key?(:dual)).to be(true)
+    expect(heat[:spt].nil?).to be(true)
+    expect(cool[:spt].nil?).to be(true)
+    expect(heat[:dual]).to be(false)
+    expect(cool[:dual]).to be(false)
+
+    expect(TBD.plenum?(space, loops, setpoints)).to be(false)
+
+    # Replace "attic" space type with "plenum", then try again.
+    attic  = model.getSpaceByName("attic")
+    expect(attic.empty?).to be(false)
+    attic  = attic.get
+    sptype = attic.spaceType
+    expect(sptype.empty?).to be(false)
+    sptype.get.setName("plenum")
+    expect(TBD.plenum?(attic, loops, setpoints)).to be(true) # works ...
```

Comment:
A first illustration: **without** valid heating/cooling setpoints in the model, a simple switch of the attic space type (to "plenum") allows an _unconditioned_ attic to be identified by TBD as an _indirectly conditioned_ plenum.